### ISSUE TITLE
How to split larger custom post types into multiple records?

### ISSUE THREAD

**Issue by ClaudiuCeia** [algolia/algoliasearch-wordpress#718]
### What did you expect to happen?
I'm hitting the 10kb on some custom types, and also on a couple of pages which use custom fields. I understand why the limit is set, I'm just unsure on how I can create multiple records for the same post using `add_filter('algolia_searchable_post_shared_attributes', $callback, [...])`.

I was expecting for the splitting to be either:
- Done automagically
- Done based on the content field? 

### What happened instead?
Getting the "over 10kb" error

### How can we reproduce this behavior?
I can explain what I'm doing:

```
$algoliaAttributesCallback = function(array $attributes, WP_Post $post) {
  return IndexCustomFields::get($attributes, $post)->index();
};

add_filter('algolia_searchable_post_shared_attributes', $algoliaAttributesCallback, 10, 2);
```

The `IndexCustomFields` is simply a factory which returns a different class depending on post type, and `index()` returns an `array` with my custom attributes which are queried from the post. 

The custom posts I'm talking about all have very clear ways to split, and I would love to be able to do it myself, but I'm not sure how:

- I can't seem to be able to append/overwrite my own "content" field to `$attributes`. From what I gathered from the docs, the `content` field is already split by the engine so I was hoping I can trick it to split my custom content too.

- I tried completely ignoring `$attributes` and setting a fresh `array`. I can set a `content` attribute like this, but it's still a problem if the field contents go over 10kb

Basically, since I have one callback per post, and the callback has to return an array of attributes, I don't see how I could create multiple records for each custom post type...

Can you provide a link to a page which shows this issue?
😨 nope

### Technical info
* WordPress version: 4.9.4
* Algolia Search plugin version: 2.9.3

**Comment by ClaudiuCeia** [comment]
So I found this: https://github.com/algolia/algoliasearch-wordpress/blob/b139d48772377e0b5b0db6356a88f953013a345d/includes/indices/class-algolia-searchable-posts-index.php#L82

After seeing this: https://github.com/algolia/algoliasearch-wordpress/issues/680#issuecomment-338678238

I tested locally and `add_filter('algolia_searchable_post_content', $cb);` could almost be a solution, but how am I going to make use of this if I only get the post content as a parameter to my function? I'm not using the native WordPress content field at all, but either way, I should at least get a post ID, no?

**Comment by ClaudiuCeia** [comment]
Sorry for rushing into this, this thing came up as a blocker right before a big project launch and I really need to find a fix - hence the PR.

**Comment by rayrutjes** [comment]
@ClaudiuCeia sorry for the delay in the processing of your request.
Your PR has been merged in #721 and should be released today or tomorrow.
I'll leave this open until then.
Can you confirm that new PR will solve your issue?

**Comment by rayrutjes** [comment]
Please checkout the 2.10.0 version of the plugin and let us know if you manage to achieve what you had in mind.

Feel free to re-open if needed!

**Comment by claytoncollie** [comment]
@ClaudiuCeia can you share how you split out a single post into multiple objects? 

I'm guessing you used this filter to grab certain parts of the post object then set a distinct flag to the post ID. 

Thanks for the help!
